tests/test_mock_sampler.py

```diff
@@ -216,7 +216,79 @@ def test_yield_arguments(self):
                                    broken_edges=delete_edges)
         self.assertTrue(len(sampler.nodelist)==4)
         self.assertTrue(len(sampler.edgelist)==1)
-        
+
+    def test_custom_mock_sampler(self):
+        """Test that MockDWaveSampler uses the provided custom mocking_sampler."""
+
+        # Define a custom sampler that always returns the same sample
```

Comment:
Perhaps we could use a more descriptive name, e.g., `ConstantSampler` or `OnesSampler`?

Reply:
can we write this test with existing samplers instead of a bespoke sampler defined here?

Reply:
> can we write this test with existing samplers instead of a bespoke sampler defined here?

None of the existing samplers are guaranteed to return an excited state. A constant sampler is nice for testing.

Reply:
how about constructing the BQM so that steepest greedy is guaranteed to return an excited state?